### PR TITLE
Replace await from all session/offering-management related component templates

### PR DESCRIPTION
**Pull request by stopfstedt**
refs https://github.com/ilios/ilios/issues/4651

### CODE REVIEW

File: addon/components/offering-manager.js [hunk 1]
```diff
@@ -12,11 +12,19 @@ export default class OfferingManagerComponent extends Component {
   @tracked showRemoveConfirmation = false;
   @tracked hoveredGroups = [];
 
+  foo = undefined;
```
Comment (jrjohnson) [inline addon/components/offering-manager.js:15]: ❓

File: addon/components/offering-manager.js [hunk 2]
```diff
@@ -12,11 +12,19 @@ export default class OfferingManagerComponent extends Component {
...
+
   @use learnerGroups = new ResolveAsyncValue(() => [this.args.offering.learnerGroups]);
   @use session = new ResolveAsyncValue(() => [this.args.offering?.session]);
   @use course = new ResolveAsyncValue(() => [this.session?.course]);
+  @use courseStartDate = new ResolveAsyncValue(() => [this.course.startDate]);
+  @use courseEndDate = new ResolveAsyncValue(() => [this.course.endDate]);
```
Comment (jrjohnson) [inline addon/components/offering-manager.js:21]: `start/endDate` on course aren't async. These can be getters
`get courseStartDate(){ return this.course?.startDate; }`
or probably even better just be `{{this.course.startDate}}` in the template.

File: addon/components/session-offerings.hbs [hunk 3]
```diff
@@ -16,8 +16,8 @@
     <NewOffering
       @session={{@session}}
       @cohorts={{this.cohorts}}
-      @courseStartDate={{await @session.course.startDate}}
-      @courseEndDate={{await @session.course.endDate}}
+      @courseStartDate={{this.courseStartDate}}
+      @courseEndDate={{this.courseEndDate}}
```
Comment (jrjohnson) [inline addon/components/session-offerings.hbs:20]: same as above on `course.startDate`

File: addon/components/sessions-grid-offering.hbs [hunk 4]
```diff
@@ -5,131 +5,124 @@
...
-        @scrollToBottom={{false}}
+  {{#if this.cohortsLoaded}}
+    {{#if (and this.isEditing)}}
+      <td colspan={{if @canUpdate "10" "9"}} class="expanded-offering-manager">
+        <OfferingForm
+          @showRoom={{true}}
+          @showInstructors={{true}}
+          @cohorts={{this.cohorts}}
+          @courseStartDate={{this.courseStartDate}}
+          @courseEndDate={{this.courseEndDate}}
```
Comment (jrjohnson) [inline addon/components/sessions-grid-offering.hbs:16]: Same on `course.start/endDate`

**Review by jrjohnson — CHANGES_REQUESTED** [review]

File: addon/components/offering-manager.js (hunk 2 above)
Comment (stopfstedt) [reply addon/components/offering-manager.js]: :heavy_check_mark:

File: addon/components/session-offerings.hbs (hunk 3 above)
Comment (stopfstedt) [reply addon/components/session-offerings.hbs]: :heavy_check_mark:

File: addon/components/sessions-grid-offering.hbs (hunk 4 above)
Comment (stopfstedt) [reply addon/components/sessions-grid-offering.hbs]: :heavy_check_mark:

**Review by jrjohnson — APPROVED** [review]

**Pull request merged**